Directory.Build.props

```
@@ -207,6 +207,9 @@
     <OutputRid Condition="'$(OutputRid)' == ''">$(PackageRID)</OutputRid>
     <OutputRid Condition="'$(PortableBuild)' == 'true'">$(_portableOS)-$(TargetArchitecture)</OutputRid>
     <TargetsLinuxBionic Condition="$(OutputRid.StartsWith('linux-bionic'))">true</TargetsLinuxBionic>
+    <TargetsLinuxMusl Condition="$(OutputRid.StartsWith('linux-musl')) or $(OutputRid.StartsWith('alpine'))">true</TargetsLinuxMusl>
+    <!-- TODO: Make distinction between AlpineLinux and LinuxMusl for Void -->
+    <TargetsAlpineLinux Condition="$(OutputRid.StartsWith('linux-musl')) or $(OutputRid.StartsWith('alpine'))">true</TargetsAlpineLinux>
```

`TargetsLinuxMusl` was enough I think; aligns with `TargetsLinuxBionic`. It covers both portable and non-portable use-cases which we care about.

---

this is what I had in mind:

```suggestion
    <TargetsAlpineLinux Condition="$(OutputRid.StartsWith('alpine'))">true</TargetsAlpineLinux>
```

we'd only use it in cases where something applies only to alpine, not all musl platforms.

---

This doesn't work as building on Alpine as portable build yields `linux-musl-$arch` for `OutputRid`. We thus need to check for non-portable RID, but AFAIK there is no variable that'll match `alpine.3.17-$arch` when building a portable build of runtime. It'll eventually be necessary to differentiate as on Void `_LinuxAbi` should not be `alpine-linux-musl`. As Void hasn't yet packaged dotnet, this is a change that is out-of-scope of this PR. I propose keeping as-is, and eventually come back to this.

---

Copy that, applying your suggestion of going back to TargetsLinuxMusl.

---

@akoeplinger Are you okay with these changes? We can double back once Void wants to target mono-based platforms. By then, I expect we'll have merged with `eng/native` anyways.

---

Discovered this today, `_portableOS` is not always `linux-musl` on Alpine. When building runtime in source-build, `RuntimeOS` is set as `alpine.3.17` which then keeps `_portableOS` as `linux` as `RuntimeOS` isn't equal to `linux-musl`. I've thus reverted to the original `TargersLinuxMusl` logics.

---

> which then keeps `_portableOS` as `linux` 

Then it is a bug and we should handle it above. https://github.com/dotnet/runtime/blob/969eaf85c5075dda9ea6241b61d42ba92f9dc9db/Directory.Build.props#L157

`'$(_runtimeOS)' == 'linux-musl' or $(_runtimeOS.StartsWith('alpine'))`

---

Copy that, thanks :)